### PR TITLE
[SERVICES-2577] Staking farms filter by depleted rewards

### PR DESCRIPTION
**Pull request by mad2sm0key**
## Reasoning
- The front-end needs a way to filter out staking farms with depleted rewards on the `filteredStakingFarms` query

  
## Proposed Changes
- add `rewardsDepleted` boolean filter to `filteredStakingFarms` query

## How to test
- the query below should only return staking farms with remaining rewards
```
query {
  filteredStakingFarms(
    filters:{
      rewardsDepleted: false
    }
    pagination:{
      first:1
    }
    sorting: {
      sortField: TVL
      sortOrder: DESC
    }
  ) {
    edges {
      cursor
      node {
        address
        apr
        farmingToken {
          identifier
        }
        farmToken {
          ticker
        }
      }
    }
    pageData {
      count
      limit
      offset
    }
  }
}
```

### CODE REVIEW

**Review by gabrielmatei — APPROVED** [review]

File: src/modules/staking/services/staking.filtering.service.ts [hunk 1]
```diff
@@ -43,4 +41,36 @@ export class StakingFilteringService {
...
+        stakeAddresses: string[],
+    ): Promise<string[]> {
+        if (
+            typeof stakingFilter.rewardsDepleted === 'undefined' ||
+            stakingFilter.rewardsDepleted === null
+        ) {
+            return stakeAddresses;
+        }
+
+        const allProduceRewardsEnabled =
```
Comment (cfaur09) [inline src/modules/staking/services/staking.filtering.service.ts:56]: you are awaiting each service call one after another,  you could parallelize these service calls using Promise.all to improve performance

File: src/modules/staking/models/staking.args.ts [hunk 2]
```diff
@@ -39,6 +39,8 @@ export class ClaimRewardsWithNewValueArgs extends GenericStakeFarmArgs {
 export class StakingFarmsFilter {
     @Field(() => String, { nullable: true })
     searchToken?: string;
+    @Field({ nullable: true })
+    rewardsDepleted?: boolean;
```
Comment (cfaur09) [inline src/modules/staking/models/staking.args.ts:43]: suggestion: `rewardsDepleted` -> `rewardDepletedPerFarm` ? 🤔

**Review by cfaur09 — APPROVED** [review]

File: src/modules/staking/services/staking.service.ts [hunk 3]
```diff
@@ -106,6 +115,37 @@ export class StakingService {
         });
     }
 
+    async getAllAccumulatedRewards(
```
Comment (claudiulataretu) [inline src/modules/staking/services/staking.service.ts:118]: I would move these methods to abi service. It should be at the same level with the single address methods

File: src/modules/staking/models/staking.args.ts [hunk 4]
```diff
@@ -39,6 +39,8 @@ export class ClaimRewardsWithNewValueArgs extends GenericStakeFarmArgs {
 export class StakingFarmsFilter {
     @Field(() => String, { nullable: true })
     searchToken?: string;
+    @Field({ nullable: true })
+    rewardsDepletedPerFarm?: boolean;
```
Comment (claudiulataretu) [inline src/modules/staking/models/staking.args.ts:43]: maybe rename into `rewardsEnded`. `depleted` would means that all rewards were distributed and it wouldn't take into account the manual stop of rewards

**Review by claudiulataretu — CHANGES_REQUESTED** [review]

File: src/modules/staking/services/staking.service.ts (hunk 3 above)
Comment (mad2sm0key) [reply src/modules/staking/services/staking.service.ts]: done

File: src/modules/staking/models/staking.args.ts (hunk 4 above)
Comment (mad2sm0key) [reply src/modules/staking/models/staking.args.ts]: done

**Review by cfaur09 — APPROVED** [review]

**Pull request merged**